nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearch5.java

```diff
@@ -0,0 +1,271 @@
+            .description("All FlowFiles that cannot be written to Elasticsearch are routed to this relationship").build();
+
+    static final Relationship REL_RETRY = new Relationship.Builder().name("retry")
+            .description("A FlowFile is routed to this relationship if the database cannot be updated but attempting the operation again may succeed")
+            .build();
+
+    public static final PropertyDescriptor ID_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("el5-put-id-attribute")
+            .displayName("Identifier Attribute")
+            .description("The name of the attribute containing the identifier for each FlowFile")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.ATTRIBUTE_KEY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("el5-put-index")
+            .displayName("Index")
+            .description("The name of the index to insert into")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(NON_EMPTY_EL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("el5-put-type")
+            .displayName("Type")
+            .description("The type of this document (used by Elasticsearch for indexing and searching)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(NON_EMPTY_EL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INDEX_OP = new PropertyDescriptor.Builder()
+            .name("el5-put-index-op")
+            .displayName("Index Operation")
+            .description("The type of the operation used to index (index, update, upsert)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(NON_EMPTY_EL_VALIDATOR)
+            .defaultValue("index")
+            .build();
+
+    public static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("el5-put-batch-size")
+            .displayName("Batch Size")
+            .description("The preferred number of FlowFiles to put to the database in a single transaction")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("100")
+            .expressionLanguageSupported(true)
+            .build();
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_RETRY);
+        return Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
```

Comment:
Same comments about doing it in static initializer (applies to both _ getRelationships_ and _ getSupportedPropertyDescriptors_).